src/components/app.jsx

```diff
@@ -23,38 +24,40 @@ import { useClient } from 'cozy-client'
 import { useI18n } from 'cozy-ui/transpiled/react/I18n'
 
 const manifest = require('../../manifest.webapp')
+import { AppRoutes } from 'constants/routes'
 import { List, Editor, Unshared } from 'components/notes'
+import { fetchIfIsNoteReadOnly } from 'lib/utils'
+import { getDataOrDefault } from 'lib/initFromDom'
 import { getReturnUrl, getSharedDocument } from 'lib/utils'
 import { useFlagSwitcher } from 'lib/debug'
-import { getDataOrDefault } from 'lib/initFromDom'
-import { fetchIfIsNoteReadOnly } from 'lib/utils'
-import { Routes } from 'constants/routes'
 
-const RoutedEditor = withRouter(props => {
+const RoutedEditor = () => {
+  const { id } = useParams()
   const returnUrl = getReturnUrl()
 
-  return (
-    <Editor
-      noteId={props.match.params.id}
-      returnUrl={returnUrl}
-      readOnly={false}
-    />
-  )
-})
+  return <Editor noteId={id} returnUrl={returnUrl} readOnly={false} />
+}
 
 const PrivateContext = () => {
   const location = useLocation()
-  const background = location.state?.background
+  const navigate = useNavigate()
+  const state = location.state
 
   return (
     <>
-      <Switch>
-        <Route path="/n/:id" component={RoutedEditor} />
-        <Route path="/" component={List} />
-      </Switch>
-
-      {background && (
-        <Route path={`/${Routes.ShareFromList}`} component={ShareModal} />
+      <Routes location={state?.backgroundLocation || location}>
+        <Route path={AppRoutes.Root} element={<List />} />
+
+        <Route path={AppRoutes.Editor} element={<RoutedEditor />} />
+      </Routes>
+
+      {state?.backgroundLocation && (
+        <Routes>
+          <Route
+            path={AppRoutes.ShareFromList}
+            element={<ShareModal onClose={() => navigate(-1)} />}
```

Comment:
maybe it should go back to root with a replace so you can't go back to modal with the back button. Any thought @Crash-- ?